### PR TITLE
Add recipient gadget

### LINKED ISSUE #188: circuits: Rename `skeleton_hash` to `payload_hash`
#### Summary
Rename `skeleton_hash` to `payload_hash` to reflect the latest changes in the paper.

### LINKED ISSUE #197: Add Recipient gadget
#### Summary
We need to implement the recipient gadget, and make the TxCircuit to use it. This needs to be implemented according to the specs of Phoenix 2.0.

### PR DESCRIPTION
**Pull request by xevisalle**
Resolves: https://github.com/dusk-network/phoenix/issues/197

### CODE REVIEW

File: circuits/src/transaction.rs [hunk 1]
```diff
@@ -184,22 +185,20 @@ impl TxOutputNote {
 ///    deposit, where a deposit refers to funds being transfered to a contract.
 ///
 /// The gadget appends the following public input values to the circuit:
-/// - `skeleton_hash`
 /// - `root`
```
Comment (marta-belles) [inline circuits/src/transaction.rs:188]: Is maybe `payload_hash` missing?

File: circuits/src/transaction.rs [hunk 2]
```diff
@@ -395,33 +398,43 @@ impl<const H: usize, const I: usize> TxCircuit<H, I> {
...
 }
 
 impl<const H: usize, const I: usize> Circuit for TxCircuit<H, I> {
     fn circuit(&self, composer: &mut Composer) -> Result<(), Error> {
-        gadget::<H, I>(
+        // Make the payload hash a public input of the circuit
+        let payload_hash = composer.append_public(self.payload_hash);
+
+        // Nullify all the tx input notes
+        nullify_gadget::<H, I>(
```
Comment (marta-belles) [inline circuits/src/transaction.rs:435]: Why is this gadget called `nullify`..?

File: circuits/src/transaction.rs (hunk 1 above)
Comment (xevisalle) [reply circuits/src/transaction.rs:188]: it's appended in the TxCircuit,  not the gadget.

File: circuits/src/transaction.rs (hunk 2 above)
Comment (xevisalle) [reply circuits/src/transaction.rs:435]: to be a bit more self-explanatory within the crate. Do you have another proposal? tx_gadget? maybe?

Comment (marta-belles) [reply circuits/src/transaction.rs:435]: considering it is the whole transfer circuit, how about `transfer_gadget`?

**Pull request closed without merging**